### PR TITLE
Update announcements.lua to add Sapped friend , self, or enemy and added battleground announcements

### PR DESCRIPTION
**Pull request by wowluanoob**
Update announcements.lua to add Sapped friend , self, or enemy and add announcements for battlegrounds

### CODE REVIEW

**Comment by wowluanoob** [comment]
for some reason idk why SAP announce is not working :( its a good idea though atleast friend sapped is not working... maybe unit references enemies only idk

File: modules/announcements.lua [hunk 1]
```diff
@@ -109,8 +124,10 @@ end
 -- Sends an announcement
 -- Param unit is only used for class coloring of messages
 function Announcements:Send(msg, throttle, unit)
-    -- only send announcements inside arenas
-    if select(2, IsInInstance()) ~= "arena" then return end
+    -- only send announcements inside arenas and battlegrounds
+    if select(2, IsInInstance()) ~= "arena" or select(2, IsInInstance()) ~= "pvp" then return end
```
Comment (ManneN1) [inline modules/announcements.lua:128]: This will be rejected. GladiusEx is an arena addon.

Comment (ManneN1) [reply modules/announcements.lua:128]: Also, this is incorrectly formatted and will fire neither in arena nor bgs.

Comment (wowluanoob) [reply modules/announcements.lua:128]: Is it possible to add some kind of sapped announcement? My fork doesnt work idk how to edit your code

**Pull request closed without merging**